File: backend/src/main/java/me/performancereservation/global/security/oauth/user/NaverOAuth2UserInfo.java
package me.performancereservation.global.security.oauth.user;

import java.util.Map;

public class NaverOAuth2UserInfo extends OAuth2UserInfo {
    public NaverOAuth2UserInfo(Map<String, Object> attributes) {
        super(attributes);
    }

    @Override
    public String getName() {
        Map<String, Object> response = (Map<String, Object>) attributes.get("response");

        return (String) response.get("name");
    }

    @Override
    public String getEmail() {
        Map<String, Object> response = (Map<String, Object>) attributes.get("response");

        return (String) response.get("email");
    }

    @Override
    public String getOauthId() {
        Map<String, Object> response = (Map<String, Object>) attributes.get("response");

        return (String) response.get("id");
    }
}

File: backend/src/main/kotlin/me/performancereservation/global/security/oauth/user/NaverOAuth2UserInfo.kt
package me.performancereservation.global.security.oauth.user

class NaverOAuth2UserInfo(attributes: Map<String, Any>) : OAuth2UserInfo(attributes) {
    private val response: Map<*, *>? = attributes["response"] as? Map<*, *>
    override val oauthId: String get() = response?.get("id") as? String ?: ""
    override val name: String get() = response?.get("name") as? String ?: ""
    override val email: String get() = response?.get("email") as? String ?: ""
}
